shell/platform/darwin/ios/ios_surface_metal_impeller.mm

```diff
@@ -64,7 +64,7 @@
   // presented. If there is a non-Flutter UIView active, such as in add2app or a
   // presentViewController page transition, then this will cause CoreAnimation assertion errors and
   // exit the app.
-  layer.presentsWithTransaction = [[NSThread currentThread] isMainThread];
+  layer.presentsWithTransaction = YES;// [[NSThread currentThread] isMainThread];
```

Comment:
Need a way to conditionally trigger this based on the presence of platform views since we won't be able to use the main thread condition anymore.

Reply:
Done